qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,1037 @@
+
+            if qubits:
+                msg += f"on qubits {qubits} "
+
+            if sched_name:
+                msg += f"in schedule {sched_name} "
+
+            if cutoff_date:
+                msg += f"with cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        # 5) Return the most recent parameter.
+        return max(candidates, key=lambda x: x.date_time).value
+
+    def get_schedule(
+        self,
+        name: str,
+        qubits: Union[int, Tuple[int, ...]],
+        assign_params: Dict[Union[str, ParameterKey], ParameterValueType] = None,
+        group: Optional[str] = "default",
+        cutoff_date: datetime = None,
+        break_parameter_coupling: bool = False,
+    ) -> ScheduleBlock:
+        """Get the template schedule with parameters assigned to values.
+
+        All the parameters in the template schedule block will be assigned to the values managed
+        by the calibrations unless they are specified in assign_params. In this case the value in
+        assign_params will override the value stored by the calibrations. A parameter value in
+        assign_params may also be a :class:`ParameterExpression`.
+
+        .. code-block:: python
+
+            # Get an xp schedule with a parametric amplitude
+            sched = cals.get_schedule("xp", 3, assign_params={"amp": Parameter("amp")})
+
+            # Get an echoed-cross-resonance schedule between qubits (0, 2) where the xp echo gates
+            # are Called schedules but leave their amplitudes as parameters.
+            assign_dict = {("amp", (0,), "xp"): Parameter("my_amp")}
+            sched = cals.get_schedule("cr", (0, 2), assign_params=assign_dict)
+
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            assign_params: The parameters to assign manually. Each parameter is specified by a
+                ParameterKey which is a named tuple of the form (parameter name, qubits,
+                schedule name). Each entry in assign_params can also be a string corresponding
+                to the name of the parameter. In this case, the schedule name and qubits of the
+                corresponding ParameterKey will be the name and qubits given as arguments to
+                get_schedule.
+            group: The calibration group from which to draw the parameters. If not specified
+                this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+            break_parameter_coupling: Boolean which when True will cause parameter couplings to
```

Comment:
I cannot understand how (when) this works. Do we want to ignore parameter assignment though we specify some values to assign_params?

Reply:
This has been changed. See e.g. here: 1dd07e8 and here https://github.com/wshanks/qiskit-experiments/tree/calibrations_definition